src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/ExtensibleSourceRepositoryProvider.cs

```diff
@@ -54,17 +57,26 @@ public ExtensibleSourceRepositoryProvider()
 
         private void EnsureInitialized()
         {
-            LazyInitializer.EnsureInitialized(ref _packageSourceProvider,
-                ref _initialized,
-                ref _lockObj,
-                () =>
-                    {
-                        #pragma warning disable CS0618 // Type or member is obsolete
-                        IPackageSourceProvider packageSourceProvider = new PackageSourceProvider(_settings.Value, enablePackageSourcesChangedEvent: true);
-                        #pragma warning restore CS0618 // Type or member is obsolete
-                        packageSourceProvider.PackageSourcesChanged += ResetRepositories;
-                        return packageSourceProvider;
-                    });
+            // this check is to avoid switching to a background thread if settings were already initialized
+            if (!Volatile.Read(ref _initialized))
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
```

Comment:
This line makes the entire fix ineffective. You're still blocking the calling thread as you were before, and you've done nothing to change which thread is calling it. The fact that within the JTF.Run delegate you switch to a background thread just means you're now using/blocking *two* threads where you used to just use one.

To fix this you either need to change which thread is blocked by changing the thread that the *caller* is on, or you need to make this an async initialization so that *no* thread is blocked while you switch to a background thread.